### PR TITLE
common code cleanup part 5

### PR DESCRIPTION
**Pull request by commy2**
- [x] LSD

### CODE REVIEW

File: addons/common/functions/fnc_debug.sqf [hunk 1]
```diff
@@ -29,13 +29,7 @@ _defaultLogDisplayLevel = [GVAR(LOGDISPLAY_LEVEL), DEFAULT_TEXT_DISPLAY] select
...
     private ["_prefix", "_message"];
 
-    switch (_level) do {
-        case 0: {_prefix = "Error"};
-        case 1: {_prefix = "Warn"};
-        case 2: {_prefix = "Debug"};
-        case 3: {_prefix = "Info"};
-        default {_prefix = "Unknown"};
-    };
+    _prefix = ["Error", "Warn", "Debug", "Info"] select (_level min 3);
```
Comment (jokoho48) [inline addons/common/functions/fnc_debug.sqf]:
```
_prefix = if (_level in [0,1,2,3]) then {
    ["Error", "Warn", "Debug", "Info"] select (_level min 3)
}; else {
    "Unknown"
};
```

Comment (commy2) [inline addons/common/functions/fnc_debug.sqf]: You are right.

That looks so dumb though. Should've left the switch tbh.

File: addons/common/functions/fnc_map.sqf [hunk 2]
```diff
@@ -12,18 +11,15 @@
...
-_array = + _this select 0;
-_code = _this select 1;
+params ["_array", "_code"];
 
-if (isNil "_array") exitWith {
-    ACE_LOGERROR_1("No array for function map in %1.",_fnc_scriptNameParent);
-    []
-};
+// copy array to not alter the original one
+_array = + _array;
```
Comment (jokoho48) [inline addons/common/functions/fnc_map.sqf]: could you check if `_array = +_array;` is realy needed if you use params

File: addons/common/functions/fnc_moduleLSDVehicles.sqf [hunk 3]
```diff
@@ -32,16 +36,21 @@ if !(_activated) exitWith {};
...
+
+            while {true} do {
                 for "_i" from 0 to (_hSCount - 1) do {
-                    _vehicle setObjectTexture [_i, (_colors select _index)];
+                    _vehicle setObjectTexture [_i, _colors select _index];
                 };
+
                 _index = (_index + 1) % 7;
+
                 sleep 0.02;
```
Comment (jokoho48) [inline addons/common/functions/fnc_moduleLSDVehicles.sqf]: PFH?

File: addons/common/functions/fnc_onAnswerRequest.sqf [hunk 4]
```diff
@@ -1,19 +1,23 @@
...
+ * N/A
  *
- * @Arguments: []
- * @Return:
- * @PublicAPI: false
+ * Arguments:
+ * ?
+ *
+ * Return Value:
+ * ?
```
Comment (jokoho48) [inline addons/common/functions/fnc_onAnswerRequest.sqf]: @Glowbal

File: addons/common/functions/fnc_map.sqf (hunk 2 above)
Comment (commy2) [inline addons/common/functions/fnc_map.sqf]: It is.

File: addons/common/functions/fnc_moduleLSDVehicles.sqf (hunk 3 above)
Comment (commy2) [inline addons/common/functions/fnc_moduleLSDVehicles.sqf]: Yes. Planned.

File: addons/common/functions/fnc_map.sqf (hunk 2 above)
Comment (commy2) [inline addons/common/functions/fnc_map.sqf]: A = [1]; [A] call {params ["_b"]; _b set [0, 2]; _b}; A
-> [2]

File: addons/common/functions/fnc_debug.sqf (hunk 1 above)
Comment (commy2) [inline addons/common/functions/fnc_debug.sqf]: fastest one:
`["Unknown", "Error", "Warn", "Debug", "Info"] select (([0, 1, 2, 3] find 4) + 1)`

don't blame me for shitty readability. I'm going with that.

**Comment by commy2** [comment]
```
 2:02:56 Error in expression < from 0 to (_hSCount - 1) do {
_vehicle setObjectTexture [_i, _colors select ace>
 2:02:56   Error position: <setObjectTexture [_i, _colors select ace>
 2:02:56   Error Generic error in expression
 2:02:56 File z\ace\addons\common\functions\fnc_moduleLSDVehicles.sqf, line 43
```

**Comment by jokoho48** [comment]
yeah i have params realy

**Comment by commy2** [comment]
Pardon?

**Comment by commy2** [comment]
Fixed. Ready to merge.
Will continue with this later tomorrow.

**Comment by commy2** [comment]
^ allows for changing GVAR(LSD_Colors) on the fly.

**Pull request merged**